pkg/client/client.go

```diff
@@ -73,6 +85,8 @@ func (c *Client) fromImageURL(imageURL string) (ImageClient, string, string) {
 	host, path := split[0], split[1]
 
 	switch {
+	case c.selfhosted.URL != "" && c.selfhosted.IsHost(host):
```

Comment:
We can follow up with another PR. It would be nice to have selfhosted be the fall back and best effort, when we can't determine the host and is also not a DockerHub shorthand.